scRNA-seq-advanced/03-dataset_integration.Rmd

```diff
@@ -649,144 +648,138 @@ scater::plotReducedDim(merged_sce,
 
-```{r fastmnn umap celltypes faceted}
+```{r fastmnn umap celltype faceted}
 scater::plotReducedDim(merged_sce,
                        dimred = "fastMNN_UMAP",
-                       # color by broad celltypes
                        colour_by = "celltype_broad",
                        point_size = 0.5,
-                       point_alpha = 0.2, 
-                       other_fields = "library") +
+                       point_alpha = 0.2,
+                       # Allow for faceting by a variable using `other_fields`:
+                       other_fields = "celltype_broad") +
   guides(colour = guide_legend(override.aes = list(size = 3, alpha = 1))) +
   ggtitle("UMAP after integration with fastMNN") +
-  facet_wrap(~library)
+  # Facet by celltype_broad
+  facet_wrap(vars(celltype_broad)) +
+  # Use a theme with background grid to more easily compare panel coordinates
+  theme_bw() 
 ```
 
-Now is a good time to remember we actually have two cell type variables: `celltype_broad` and `celltype_fine`, where the latter contains finer-grained Tumor characterizations based on differentiation stage.
+All `Tumor` cell types appear to group with one another, with the main exception of `Tumor_Mesoderm` (along with what looks like one `Tumor_Myoblast` cell) for which a small number of cells are located with healthy muscle tissue.
+This _potentially_ suggests that these Tumor cells were incorrectly classified and are actually not cancerous, which is a hypothesis one could further explore, but on the other hand RMS does share many features with muscle tissue.
 
-Let's see what we observe if we color the UMAP instead by the finer-grained cell types.
-For this plot, we'll focus only on Tumor cell types and remove all the "healthy" cells so we can see Tumor cells more clearly:
+### Integration with `harmony`
 
-```{r fastmnn umap fine tumor}
-# First, let's filter to only cell types that contain the word "Tumor"
-# In which cells is "Tumor" present in the celltype_fine variable?
-tumor_cells <- grep("Tumor", merged_sce$celltype_fine)
-merged_sce_tumor <- merged_sce[,tumor_cells]
+`fastMNN` is only one of many approaches to perform integration, and different methods have different capabilities and are not unlikely to give different results.
+For example some methods can accommodate additional covariates (e.g., technology, patient, diagnosis, etc.) that may influence integration. 
+In fact the data we are using has a known _patient_ covariate; `SCPCL000479` and `SCPCL000480` are from the first patient, and `SCPCL000481` and `SCPCL000482` are from the second patient.
 
+So, let's perform integration with a method that can use this information - [`harmony`](https://portals.broadinstitute.org/harmony/)!
+One such method is , which (algorithm aside) has a couple practical differences from `fastMNN`:
```

Comment:
This sentence is confusing to me. I think the beginning part is unfinished?

Reply:
Ah, it should include the word `harmony` 😂

Reply:
Then these two sentences are redundant? Or at least the "one such method" part...
```suggestion
Algorithm aside, `harmony` has a few practical differences from `fastMNN`:
```

Reply:
Ah yes, this sentence should just go away. Revision problems.